### PR TITLE
simplify room mechanism

### PR DESCRIPTION
**Pull request by BeniRupp**
Add generic mechanism for locked room doors and room intros.
A task can have a timer, now.

### CODE REVIEW

File: softwarebude/escape_room.py [hunk 1]
```diff
@@ -10,33 +9,21 @@ def __init__(self, name, description):
...
+            "hello world",
+            30,
         )
         print("Oh. Hallo. Gut das du da bist. Ich brauche deine Hilfe!")
         helper.wait(2)
         print("Ich komme bei diesem Problem nicht weiter. Kannst du es für mich lösen?")
         helper.wait(2)
         print(
-            "Eigentlich hast du keine anderer möglichkeit. Dafür bezahle ich dich schließlich"
+            "Eigentlich hast du keine anderer Möglichkeit. Dafür bezahle ich dich schließlich."
```
Comment (RSchlenker) [inline softwarebude/escape_room.py]: Fix Typo "keine andere Möglichkeit"

File: softwarebude/task.py [hunk 2]
```diff
@@ -1,10 +1,41 @@
...
+                self.stop_counter()
+                return True
+            elif self.time_to_answer():
+                print("Das war nicht korrekt.")
+
+    def time_to_answer(self):
+        if self.countdown is not None:
+            return not self.countdown.countdown_expired
+        else:
+            return True
```
Comment (RSchlenker) [inline softwarebude/task.py]: Maybe we can use a more expressive method name here? Like: is_time_to_answer_left(self) or answer_time_left() or still_time_to_answer()

File: softwarebude/treasure_room.py [hunk 3]
```diff
@@ -41,15 +41,4 @@ def intro_treasure_room(self):
...
-        self.task = Task("Gib Code ein: ", code)
-        print("Dieser Raum ist verriegelt.")
-        print('Um zurück zu gehen, gebe "Exit" ein')
-        while True:
-            answer = input(self.task.show_question())
-            if self.task.is_answer_correct(answer):
-                return True
-            if answer == "exit" or answer == "Exit":
-                return False
+        print("...eine Gehaltserhöhung! Du bekommst mehr Geld. 💸")
```
Comment (RSchlenker) [inline softwarebude/treasure_room.py:44]: Are these icons unix compatible?

**Review by RSchlenker — CHANGES_REQUESTED** [review]

File: softwarebude/treasure_room.py (hunk 3 above)
Comment (BeniRupp) [reply softwarebude/treasure_room.py:44]: You can add an issue for that. 🙂

**Review by RSchlenker — APPROVED** [review]

**Pull request merged**